InscryptionAPI/Ascension/ChallengeManager.cs

```diff
@@ -4,56 +4,187 @@
 using InscryptionAPI.Guid;
 using InscryptionAPI.Helpers;
 using System.Collections.ObjectModel;
+using InscryptionAPI.Boons;
+using System.Collections;
 
 namespace InscryptionAPI.Ascension;
 
 [HarmonyPatch]
 public static class ChallengeManager
```

Comment:
The usage of the updated API needs to be documented - **especially** the parts about dependent/incompatible challenges. And you can apparently have challenges that aren't displayed on the screen - what is the use case for that? That sort of thing.

Reply:
hopefully done